src/lib/install.sh

```diff
@@ -21,6 +22,63 @@ install_single() {
     ln -fs "$_LOCAL/opt/$2/current/$path_to_bin" "$_LOCAL/bin/$2"
 }
 
+install_multi() {
+    #: Used to install multi-binary recipes
+    #:    ($1) tarball_path - path to the source tarball
+    #:    ($2) recipe_name - the name of the recipe currently being installed
+    #:    ($3) version - the version of the recipe currently being installed
+    local tarball_path="$1"; local recipe_name="$2"; local version="$3"
+
+    #: create proper .local/opt/name/version directory structure and extract the tarball
+    local recipe_dir="$_LOCAL/opt/$recipe_name"
+    if [ -e "$recipe_dir/$version" ]; then
+        _warn "Current version seems to have been installed in $recipe_dir"
+        # exit here?
```

Comment:
I would say **yes**, we should exit here with a non-zero code (e.g. 9 for now to avoid collisions with lower values which may mean something else). Also, maybe adding something like "please remove it and try again".